plugins/official/stack-snippets/src/snippet-view.ts

```diff
@@ -184,7 +158,71 @@ export class StackSnippetView implements NodeView {
 
         // Update the result container if metadata has changed
         const content = this.contentNode;
-        if (metaChanged && content) {
+
+        //Show the results, if the node meta allows it
+        if (content && node.attrs.showResult) {
+            if (!this.showButton.classList.contains("d-none")) {
+                this.showButton.classList.add("d-none");
+            }
+            if (this.hideButton.classList.contains("d-none")) {
+                this.hideButton.classList.remove("d-none");
+            }
+            if (this.resultContainer.classList.contains("d-none")) {
+                this.resultContainer.classList.remove("d-none");
+            }
+        } else if (content && !node.attrs.showResult) {
+            if (this.showButton.classList.contains("d-none")) {
+                this.showButton.classList.remove("d-none");
+            }
+            if (!this.hideButton.classList.contains("d-none")) {
+                this.hideButton.classList.add("d-none");
+            }
+            if (!this.resultContainer.classList.contains("d-none")) {
+                this.resultContainer.classList.add("d-none");
+            }
+        }
+
+        //Fullscreen the results, if the node meta needs it
+        if (content && node.attrs.fullscreen) {
+            if (!this.dom.classList.contains("snippet-fullscreen")) {
+                //We use `.snippet-fullscreen` as a marker for the rest of the styling
+                this.dom.classList.add("snippet-fullscreen");
```

Comment:
NIT: I think you can simplify a bit this code, by passing more than one class at once
```
this.dom.classList.add("class1", "class2", etc...
```
Same for remove. You could even put them in an array and spread it. 
https://developer.mozilla.org/en-US/docs/Web/API/Element/classList#examples